src/main/java/io/tus/java/client/TusUploader.java

```diff
@@ -176,7 +177,15 @@ public int getRequestPayloadSize() {
     public int uploadChunk() throws IOException, ProtocolException {
         openConnection();
 
-        int bytesToRead = Math.min(getChunkSize(), bytesRemainingForRequest);
+        final int bytesToRead;
+        if (client.chunkedTransferEncoding())
+            bytesToRead = Math.min(getChunkSize(), bytesRemainingForRequest);
+        else {
+            // No chunked tranfer encoding, so we upload a single chunk.
+            bytesToRead = getChunkSize();
```

Comment:
I think there is still some confusion about the chunk size vs payload size. Traditionally the payload size determines how big the body of a HTTP can get. The chunk size on the other hand says, how much data is read from the input stream and written to the TCP connection in one iteration. Usually, you write multiple chunks per HTTP request, so the payload size is bigger than the chunk size.

However, if I understand this correctly, the chunk size now determines how big a HTTP request gets and the payload size is not used anymore. This would be contrary to how tus-java-client behaves when chunked transfer is enabled.

Reply:
I did the opposite reasoning:
chunk size exists because something in the chain does not support more than that. So, when chunk transfer is not supported, it seems better to use TUS protocol in place of chunk transfer and then, send many complete HTTP+TUS messages based on chunk size.

Furthermore, as a dumb user, I only wish things work without having to fine tune chunk size or payload size, depending on the server's capabilities. I like *automagical* approach. :-)

But I'm completly new to all of this. So I don't know what sort of logic is expected.

Reply:
> chunk size exists because something in the chain does not support more than that. 

That's not entirely true. The reason for chunk size is that the user can control how much data is read from the input source in one go. This is important to control how big the in-memory buffer of tus-java-client is, in order to avoid excess memory usage.

You are referring to the payload size when you say "something [e.g. a proxy] in the chain does not support more than that". Does that make sense?

> Furthermore, as a dumb user, I only wish things work without having to fine tune chunk size or payload size, depending on the server's capabilities. I like automagical approach. :-)

Yes, I would like that as well and it will likely come in the next major release but not right now :)

Reply:
> This is important to control how big the in-memory buffer of tus-java-client is, in order to avoid excess memory usage.

OK, so you state that client (memory) is "something in the chain does not support more than that", first part of the chain.